# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/farida596/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

**Lane:** Refresh / Content Opportunity Scoring

**Research Question:** Which webpages should be prioritized for content refresh to improve their search performance?

I chose the Refresh / Content Opportunity Scoring lane because I want to identify the webpages that are most likely to benefit from a content review and refresh based on their historical search and engagement performance. Since large websites contain many pages, reviewing each page manually is time-consuming. This project aims to rank pages by their refresh opportunity, helping the SEO team prioritize their efforts and focus on the pages with the highest potential for improvement.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## What decision does your work improve?

My work improves the decision of which webpages should be prioritized for content refresh. The goal is to identify the pages that need to be reviewed first, ensuring that high-priority pages are not overlooked while avoiding spending time refreshing pages that do not currently need updates.

## Who acts on it?

The SEO and content teams use the ranked recommendations to decide which pages should be reviewed and refreshed first. This helps them prioritize their work and make better use of their time and resources.

## What does a wrong recommendation cost?

A wrong recommendation could cause the team to spend time and resources refreshing pages that do not need updates while overlooking pages that require attention the most. As a result, valuable opportunities for improving search performance may be missed.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [11]:
!git clone https://github.com/farida596/flyrank-ml-internship.git

fatal: destination path 'flyrank-ml-internship' already exists and is not an empty directory.


In [12]:
!find flyrank-ml-internship -name "*.csv"


flyrank-ml-internship/outputs/refresh_queue_sample.csv
flyrank-ml-internship/data/raw/content_refresh_anonymized.csv


In [13]:
import pandas as pd
# Load the dataset
df = pd.read_csv("flyrank-ml-internship/data/raw/content_refresh_anonymized.csv")
# Display the first five rows
df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


### Explore the dataset

Next, I inspect the available columns and calculate a few summary statistics to better understand the dataset and support my choice of the Refresh / Content Opportunity Scoring lane.

In [14]:
# Display all column names
list(df.columns)

['content_id',
 'client_id',
 'search_volume',
 'competition',
 'competition_level',
 'cpc',
 'content_type',
 'main_intent',
 'word_count',
 'char_count',
 'provider_used',
 'model_used',
 'impressions_90d',
 'clicks_90d',
 'pageviews_90d',
 'sessions_90d',
 'users_90d',
 'engaged_sessions_90d',
 'ai_sessions_90d',
 'scroll_events_90d',
 'days_with_impressions',
 'days_with_sessions',
 'impressions_last_30d',
 'clicks_last_30d',
 'sessions_last_30d',
 'impressions_prev_30d',
 'clicks_prev_30d',
 'sessions_prev_30d',
 'content_age_days',
 'age_tier',
 'age_tier_order',
 'days_since_last_update',
 'freshness_tier',
 'word_count_tier',
 'char_count_tier',
 'ctr',
 'avg_position',
 'engagement_rate',
 'scroll_rate',
 'ai_traffic_pct',
 'impression_tier',
 'position_tier',
 'trend_direction',
 'trend_pct']

### Quick look at the data

To support my choice of the **Refresh / Content Opportunity Scoring** lane, I explored the starter dataset and calculated a few summary statistics. These statistics provide an overview of the dataset and help explain why prioritizing pages for content refresh is a meaningful problem.

In [15]:
# Number of pages in the dataset
print("Number of pages:", len(df))
# Average search performance
print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Average clicks (90d):", round(df["clicks_90d"].mean(), 2))
# Distribution of trend directions
print("\nTrend direction:")
print(df["trend_direction"].value_counts())

Number of pages: 30000
Average impressions (90d): 5200.37
Average clicks (90d): 16.1

Trend direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


The starter dataset contains **30,000 webpages**, making it impractical to manually review every page for content refresh. On average, webpages received **5,200.37 impressions** and **16.10 clicks** over the last 90 days, showing that search performance varies across pages. In addition, **16,262 pages are currently showing a downward trend**, which represents more than half of the dataset. These observations suggest that prioritizing which pages should be reviewed and refreshed first is an important decision, making the **Refresh / Content Opportunity Scoring** lane a meaningful problem to work on.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What my work can claim

My work can identify and rank the webpages that are most likely to benefit from a content refresh based on historical search and engagement data. It is designed to support decision-making by helping the SEO and content teams prioritize which pages should be reviewed first.

## What my work cannot claim

My work cannot guarantee that refreshing a page will improve its search performance, nor can it predict exactly how much a page will improve or how long it will perform well before needing another refresh. It also cannot prove that a content refresh is the direct cause of any future improvement, as many external factors can affect search performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.